### PR TITLE
Fix Cygwin launch

### PR DESCRIPTION
**Pull request by alexander-myltsev**
Closes https://github.com/lampepfl/dotty/issues/703

- Fix typo in JLINE_JAR
- Implement proper Java launch in Cygwin

### CODE REVIEW

**Comment by odersky** [comment]
@DarkDimius, @smarter Can one of you review this? I am not an expert on launch scripts.

**Comment by DarkDimius** [comment]
I do not have any experience with Cygwin, neither do I have a Windows machine to test it.

This change broke `dotc` on OS X, as the classpath does not get collected anymore. Here is execution log: https://gist.github.com/DarkDimius/34f94088dabc0ca5f3d2

**Comment by alexander-myltsev** [comment]
@DarkDimius, thanks for log. I would investigate soon.

File: bin/dotc [hunk 1]
```diff
@@ -188,11 +188,24 @@ trap onExit INT
...
+    if [[ -n $bootcp ]]; then
+      boot_classpath="$(cygpath --path --$format "$toolchain:$MAIN_JAR")"
+      classpath="$(cygpath --path --$format "$MAIN_JAR:$TEST_JAR")"
+      cpArgs="-Xbootclasspath/a:$boot_classpath -classpath $classpath"
+    else
+      classpath="$(cygpath --path --$format "$toolchain:$MAIN_JAR:$TEST_JAR")"
+      cpArgs="-classpath $classpath"
+    fi
   fi
+  echo ${cpArgs}
```
Comment (DarkDimius) [inline bin/dotc]: cpArgs is not defined in case `! -n "$cygwin"`

**Comment by alexander-myltsev** [comment]
Works on Win and Ubuntu now. @DarkDimius can you please check it on Mac?

**Comment by DarkDimius** [comment]
LGTM.

**Pull request merged**